go/service/gregor.go

```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	ibm := m.ToInBandMessage()
+	if ibm != nil {
+		return g.handleInBandMessage(ctx, ibm)
+	}
+
 	obm := m.ToOutOfBandMessage()
-	if obm == nil {
-		return fmt.Errorf("gregor handler: out-of-band message nil")
+	if obm != nil {
+		return g.handleOutOfBandMessage(ctx, obm)
+	}
+
+	return fmt.Errorf("gregor handler: both in-band and out-of-band message nil")
+}
+
+func (g *gregorHandler) handleInBandMessage(ctx context.Context, ibm gregor.InBandMessage) error {
+	g.G().Log.Debug("gregor handler: handleInBand: %+v", ibm)
+
+	sync := ibm.ToStateSyncMessage()
+	if sync != nil {
+		g.G().Log.Debug("gregor handler: state sync message")
+	}
+
+	update := ibm.ToStateUpdateMessage()
+	if update != nil {
+		g.G().Log.Debug("gregor handler: state update message")
+
+		item := update.Creation()
+		if item != nil {
+			g.G().Log.Debug("gregor handler: item created")
+
+			category := ""
+			if item.Category() != nil {
+				category = item.Category().String()
+			}
+			if category == "show_tracker_popup" {
+				return g.handleShowTrackerPopup(ctx, item)
+			}
+			g.G().Log.Errorf("Unrecognized item category: %s", item.Category())
+		}
+
+		dismissal := update.Dismissal()
+		if dismissal != nil {
+			g.G().Log.Debug("gregor handler: dismissal!!!")
+		}
+	}
+
+	return nil
+}
+
+func (g *gregorHandler) handleShowTrackerPopup(ctx context.Context, item gregor.Item) error {
+	g.G().Log.Debug("gregor handler: handleShowTrackerPopup: %+v", item)
+	if item.Body() == nil {
+		return errors.New("gregor handler for show_tracker_popup: nil message body")
+	}
+	body, err := jsonw.Unmarshal(item.Body().Bytes())
+	if err != nil {
+		g.G().Log.Error("body failed to unmarshal", err)
+		return err
+	}
+	uid, err := body.AtPath("uid").GetString()
+	if err != nil {
+		g.G().Log.Error("failed to extract uid", err)
+		return err
+	}
+
+	// Load the user with that UID, to get a username.
+	// TODO: Should we load by username@service instead? I worry that might not
+	// be unique...
+	user, err := libkb.LoadUser(libkb.LoadUserArg{UID: keybase1.UID(uid)})
```

Comment:
uid seems good to me.

Can you use `libkb.NewLoadUserByUIDArg(...)` so it it contextified?